src/Server.cpp

```diff
@@ -122,13 +123,29 @@ Server& Server::operator=(const Server& ref)
 void Server::run()
 {
 	RequestParser parser;
+	const int maxEvents = 10;
+	const int timeout = 5000; // epoll timeout in milliseconds
+	const int acceptedIdleTime = 10; // time in seconds until an idle connection is closed
 
 	while (true) {
-		struct epoll_event events[MAX_EVENTS];
+		struct epoll_event events[maxEvents];
 		// Blocking call to epoll_wait
-		int nfds = epoll_wait(m_epfd, events, MAX_EVENTS, -1);
+		int nfds = epoll_wait(m_epfd, events, maxEvents, timeout);
 		if (nfds < 0)
 			throw std::runtime_error("epoll_wait");
+		if (nfds == 0) {
+			LOG_WARN << "Timeout: no events received within " << timeout << " ms.";
+
+			time_t currentTime = std::time(NULL);
+			for (std::map<int, time_t>::iterator iter = m_connectionLastActive.begin();
+				 iter != m_connectionLastActive.end(); iter++) {
+				if (currentTime - iter->second > acceptedIdleTime) {
+					close(iter->first);
+					m_connectionLastActive.erase(iter);
+				}
+			}
+			continue;
```

Comment:
In #22 I do this check every time epoll_wait() returns, not only if no event is received. If we return from it we might as well check for timeout.

Reply:
Legit.